### PR TITLE
Support x-key usage in xliff file

### PR DESCRIPTION
**Pull request by gouniLee**
Related PR: https://github.com/iLib-js/ilib-loctool-webos-javascript/pull/3

webOS has additional xliff usage. it is `x-key` property. if xliff has it, it has to be the source.
Following cases, the source has to be `pictureModeSetting` instead `Customize`. I've updated code to match the source with x-key value in this case.

Example)
* xliff:
`<source x-key="pictureModeSetting">Customize</source>`

* js:
`<SettingsDivider>{$L({key:'pictureModeSettings', value:'Customize'})}</SettingsDivider>`

### CODE REVIEW

File: lib/Xliff.js [hunk 1]
```diff
@@ -816,6 +816,11 @@ Xliff.prototype.deserialize = function(xml) {
                             tu.resname = tu.source["$t"];
                         }
 
+                        if (tu.source["x-key"]) {
+                             tu.source["$t"]= tu.source["x-key"];
+                             tu.resname = tu.source["x-key"];
+                        }
+
```
Comment (ehoogerbeets) [inline lib/Xliff.js]: These lines should probably be moved up to line 814 instead. The lines 815-817 are the last resort/default case...

File: test/testfiles/test.xliff [hunk 2]
```diff
@@ -16,6 +16,11 @@
         <target>bebe bebe</target>
         <note annotates="source">come &amp; enjoy it with us</note>
       </trans-unit>
+      <trans-unit id="3" resname="ohhuzzah" restype="string" datatype="plaintext">
+        <source x-key="oh">baby baby</source>
+        <target>oh bebe bebe</target>
+        <note annotates="source">come &amp; enjoy it with them</note>
+      </trans-unit>
```
Comment (ehoogerbeets) [inline test/testfiles/test.xliff]: For this test, the x-key replaces the resname, right? You should have a test where the trans-unit has no resname attribute on it, as well as this test, and make sure that the resname turns out correctly.

File: lib/Xliff.js (hunk 1 above)
Comment (ehoogerbeets) [inline lib/Xliff.js]: Also, all of the LG xliffs are in 2.0 format, right? Do we really need to add this to the 1.2 xliff reader?

Maybe this x-key should be in the 2.0 branch. (BTW - that branch is pushed so you can look at it, but it is not finished yet.)

Comment (gouniLee) [reply lib/Xliff.js]: At this point, Yes. Regarding  Xliff 2.0 support it's not ready to be used. So In order to check and verify loctool in webOS,  I need this to add in xliff reader the current version.

Comment (gouniLee) [reply lib/Xliff.js]: I'll check your 2.0 branch. As I shared through email,`x-key` attribute is removed in xliff 2.0 then matches the `name` attribute in `unit` tag. such as
```
<xliff xmlns="urn:oasis:names:tc:xliff:document:2.0" version="2.0" srcLang="en-KR" trgLang="ko-KR">
 <file id="settings_f1" original="settings">
  <group id="settings_g1" name="javascript">
   <unit id="settings_140" name="pictureModeSettings">
    <segment>
     <source>Customize</source>
     <target>상세 설정</target>
    </segment>
   </unit>
...
```

File: test/testfiles/test.xliff (hunk 2 above)
Comment (gouniLee) [reply test/testfiles/test.xliff]: I've added another `trans-unit`  which doesn't have `rename` and `datatype`

File: lib/Xliff.js (hunk 1 above)
Comment (gouniLee) [reply lib/Xliff.js]: I updated it

**Pull request merged**